internal/project/deploy.go

```diff
@@ -65,7 +70,14 @@ func deploy(
 
 	}
 
-	c, err := srv.Project.Deploy(globalFlags.Network, deployFlags.Update)
+	var deployFunc services.Update
+	if deployFlags.ShowDiff {
+		deployFunc = updateWithPrompt
+	} else {
+		deployFunc = services.UpdateExistingContract(deployFlags.Update)
+	}
```

Comment:
More idiomatic way in Go is:
```suggestion
	deployFunc := services.UpdateExistingContract(deployFlags.Update)
	if deployFlags.ShowDiff {
		deployFunc = updateWithPrompt
	}
```